server.js

```diff
@@ -1,23 +1,81 @@
 const express = require("express");
 const cors = require("cors");
+const bp = require("body-parser");
+const fs = require("fs");
 
 const app = express();
+let data = JSON.parse(fs.readFileSync("data.json", "utf-8"));
+let maxID = Math.max(...data.map((c) => c.id));
+const messages = data;
 
 app.use(cors());
+app.use(bp.json());
+app.use(bp.urlencoded({ extended: true }));
 
-const welcomeMessage = {
-  id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
-};
-
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+//Creates a message and post it to the database
+app.post("/messages", function (req, res) {
+  let userName = req.body.userName;
+  let userMessage = req.body.userMessage;
+  if (userName === "" || userMessage === "") {
+    res.status(400).send("The input is invalid");
+  } else {
+    const newChat = {
+      id: ++maxID,
+      from: userName,
+      text: userMessage,
+    };
+    data.push(newChat);
+    save();
+    res.send(data);
+  }
+});
 
+//Serves the main HTML file for the home page
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+//Give all the messages in the data base
+app.get("/messages", function (req, res) {
+  res.json(data);
+});
+
+//Searches a message based on input ID
+app.get("/messages/search/", function (req, res) {
+  let userSearchedID = req.query.randomNumber;
+  if (!Number(userSearchedID)) {
+    res.send("Provide a Number, Please");
+  }
+  data.forEach((msg) => {
+    if (msg.id === Number(userSearchedID)) {
+      res.status(200).send(msg);
+    }
+  });
+  res.status(404).send("Message Does Not Exist In The DataBase");
```

Comment:
does line 54 not cause an exception to bet thrown in the server each request? The response is sent on both line 47 and on line 51, but there is no return, meaning that line 54 will be executed, causing Node to throw an exception. Adding a return statement after the responses on lines 47 and 51 to terminate the handler